liquidctl/driver/krakenz3.py

```diff
@@ -0,0 +1,676 @@
+            color = list(itertools.chain(*colors)) + [0x00, 0x00, 0x00] * (maxcolors - color_count)
+            speed_value = _SPEED_VALUE[speed_scale][sval]
+            self._hid_write([0x22, 0x10, cid, 0x00] + color)
+            self._hid_write([0x22, 0x11, cid, 0x00])
+            self._hid_write(
+                [0x22, 0xA0, cid, 0x00, mval]
+                + speed_value
+                + [0x08, 0x00, 0x00, 0x80, 0x00, 0x32, 0x00, 0x00, 0x01]
+            )
+
+        elif mode == "wings":  # wings requires special handling
+            self._hid_write([0x22, 0x10, cid])  # clear out all independent LEDs
+            self._hid_write([0x22, 0x11, cid])  # clear out all independent LEDs
+            color_lists = {}
+            color_lists[0] = colors[0] * 2
+            color_lists[1] = [int(x // 2.5) for x in color_lists[0]]
+            color_lists[2] = [int(x // 4) for x in color_lists[1]]
+            color_lists[3] = [0x00] * 8
+            speed_value = _SPEED_VALUE[speed_scale][sval]
+            for i in range(8):  # send color scheme first, before enabling wings mode
+                mod = 0x05 if i in [3, 7] else 0x01
+                alt = [0x04, 0x84] if i // 4 == 0 else [0x84, 0x04]
+                msg = (
+                    [0x22, 0x20, cid, i, 0x04]
+                    + speed_value
+                    + [mod]
+                    + [0x00] * 7
+                    + [0x02]
+                    + alt
+                    + [0x00] * 10
+                )
+                self._hid_write(msg + color_lists[i % 4])
+            self._hid_write([0x22, 0x03, cid, 0x08])  # this actually enables wings mode
+
+        else:
+            opcode = [0x2A, 0x04]
+            address = [cid, cid]
+            speed_value = _SPEED_VALUE[speed_scale][sval]
+            header = opcode + address + [mval] + speed_value
+            color = list(itertools.chain(*colors)) + [0, 0, 0] * (16 - color_count)
+
+            if "marquee" in mode:
+                backward_byte = 0x04
+            elif mode == "starry-night" or "moving-alternating" in mode:
+                backward_byte = 0x01
+            else:
+                backward_byte = 0x00
+
+            backward_byte += map_direction(direction, 0, 0x02)
+
+            if mode == "fading" or mode == "pulse" or mode == "breathing":
+                mode_related = 0x08
+            elif mode == "tai-chi":
+                mode_related = 0x05
+            elif mode == "water-cooler":
+                mode_related = 0x05
+                color_count = 0x01
+            elif mode == "loading":
+                mode_related = 0x04
+            else:
+                mode_related = 0x00
+
+            static_byte = _STATIC_VALUE[cid]
+            led_size = size_variant if mval == 0x03 or mval == 0x05 else 0x03
+            footer = [backward_byte, color_count, mode_related, static_byte, led_size]
+            self._hid_write(header + color + footer)
```

Comment:
While one might argue that having all the code for this device in this module is easier to reason about, _and they would be right in general,_ this is a significant amount of duplication, for a feature that almost doesn't belong here.  It's noise for what we care about here, and it's an ugly feature that would now need to be maintained twice.

But I don't have a good solution for it yet...

One idea might be to _encapsulate_ a subclass of `KrakenX3` and forward these methods to it, but it would require some many hacks to our probing code that it's probably not worth it.  Another idea is to add a common base class for both drivers, that only takes care of the LEDs (and, in the future, maybe hwmon support too).